core/sector_storage/impl/local_worker.cpp

```diff
@@ -270,20 +270,23 @@ namespace fc::sector_storage {
 
   outcome::result<CallId> LocalWorker::addPiece(
       const SectorRef &sector,
-      gsl::span<const UnpaddedPieceSize> piece_sizes,
+      VectorCoW<UnpaddedPieceSize> piece_sizes,
       const UnpaddedPieceSize &new_piece_size,
       proofs::PieceData piece_data) {
     return asyncCall(
         sector,
         return_->ReturnAddPiece,
-        [=, piece_data{std::make_shared<PieceData>(std::move(piece_data))}](
+        [=,
+         piece_sizes{std::make_shared<VectorCoW<UnpaddedPieceSize>>(
+             std::move(piece_sizes))},
```

Comment:
should span be allowed? eg caller must keep span contents during async call
```suggestion
         piece_sizes{std::move(piece_sizes)},
```
or should it synchronously be copied from span before async operation?
```suggestion
         piece_sizes{piece_sizes.into()},
```

Reply:
Fixed